dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -2256,22 +2268,39 @@ public Response handleGitHubInstallation(@ApiParam(hidden = true) @Parameter(hid
         });
 
         if (added) {
-            // make some educated guesses whether we should try to retrospectively release some old versions
+            // for each added repository, try to retrospectively release some old versions.
+            // if we inspected all of the branches and didn't find any that were releasable,
+            // attempt to infer/deliver a .dockstore.yml on the "most important" branch.
             // note that for large organizations, this loop could be quite large if many repositories are added at the same time
+            GitHubSourceCodeRepo gitHubSourceCodeRepo = (GitHubSourceCodeRepo)SourceCodeRepoFactory.createGitHubAppRepo(installationId);
             for (String repository: repositories) {
-                final Set<String> strings = identifyGitReferencesToRelease(repository, installationId);
-                for (String gitReference: strings) {
-                    if (LOG.isInfoEnabled()) {
-                        LOG.info(String.format("Retrospectively processing branch/tag %s in %s(%s)", Utilities.cleanForLogging(gitReference), Utilities.cleanForLogging(repository),
-                            Utilities.cleanForLogging(username)));
+                final int maximumBranchCount = 5;
+                final List<String> importantBranches = identifyImportantBranches(repository, gitHubSourceCodeRepo);
+                final List<String> releasableReferences = identifyGitReferencesToRelease(repository, installationId, subList(importantBranches, maximumBranchCount));
+                if (releasableReferences.isEmpty()) {
+                    boolean inspectedAllBranches = importantBranches.size() <= maximumBranchCount;
+                    if (inspectedAllBranches) {
+                        // Create recommended action
+                        notifyIfPotentiallyContainsEntries(Optional.of(user), repository, installationId, importantBranches);
+                    }
+                } else {
+                    for (String gitReference: releasableReferences) {
+                        if (LOG.isInfoEnabled()) {
+                            LOG.info(String.format("Retrospectively processing branch/tag %s in %s(%s)", Utilities.cleanForLogging(gitReference), Utilities.cleanForLogging(repository),
+                                Utilities.cleanForLogging(username)));
+                        }
+                        githubWebhookRelease(repository, new GitHubUsernames(username, Set.of()), gitReference, installationId, deliveryId, null, false);
                     }
-                    githubWebhookRelease(repository, new GitHubUsernames(username, Set.of()), gitReference, installationId, deliveryId, null, false);
                 }
             }
         }
         return Response.status(HttpStatus.SC_OK).build();
     }
 
+    private List<String> subList(List<String> values, int count) {
+        return values.stream().limit(count).toList();
```

Comment:
is this much different from https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html#subList(int,int) with the first parameter set to 0?